src/config/mongoose_config_spec.erl

```diff
@@ -137,11 +139,22 @@ root() ->
 host_config() ->
     #section{
        items = #{%% Host is only validated here - it is stored in the path,
-                 %%   see mongoose_config_parser_toml:item_key/1
+                 %% see mongoose_config_parser_toml:item_key/1
+                 %%
+                 %% for every configured host the host_type of the same name
+                 %% is declared automatically. As host_config section is now
+                 %% used for changing configuration of the host_type, we don't
+                 %% need host option any more. but to stay compatible with an
+                 %% old config format we keep host option as well. now it is
+                 %% just a synonym to host_type.
                  <<"host">> => #option{type = binary,
                                        validate = non_empty,
                                        format = skip},
 
+                 <<"host_type">> => #option{type = binary,
```

Comment:
**Just a thought - no need to change anything:**

There is one thought that I have to share here: if someone makes the mistake below it is difficult to tell which one is a host and which one is a host type.

```
[general]
  hosts = ["host1"]
  host_types = ["type1"]

[[host_config]]
  host = "type1"
  (...)

[[host_config]]
  host_type = "host1"
```

Maybe it would make sense to have a strict check here. it's not mandatory now of course - just a thought. If we make it mandatory some time in the future and put `{host_type, ...}` or `{host, ...}` in the path, we could avoid the whole second pass through the config done in `check_dynamic_domains_support/1`.

Reply:
I don't want to introduce that limitation, since we declare that for each and every host there's an implicit definition of the host type of the same name as host's domain name. I see `host_config.host` as a synonym for `host_config.host_type` which is necessary only for the backward compatibility purposes.

Reply:
Yes I am fine to keep it as it is.